File: Compiler/src/compiler/values/IntegerValue.java
package compiler.values;

import compiler.Value;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;

public class IntegerValue extends Value {

    public IntegerValue(Integer value) {
        super(Integer.toString(value));
    }

    @Override
    public DataType resultDataType() {
        return DataType.INTEGER;
    }

    @Override
    public CompilationResult compile() {
        return new CompilationResult(JenaUtil.genIntegerVal(value), "", "");
    }
}

File: Compiler/src/compiler/values/IntegerValue.kt
package compiler.values

import compiler.Operator
import compiler.Value
import util.CompilationResult
import util.DataType
import util.JenaUtil

class IntegerValue(value: Int) : Value(value.toString()) {

    override fun resultDataType(): DataType = DataType.Integer

    override fun compile(): CompilationResult =
        CompilationResult(JenaUtil.genIntegerVal(value), listOf(""), "")

    override fun clone(): Operator {
        return IntegerValue(value.toInt())
    }
}